test/sneaker_handlers/exponential_backoff_handler_test.rb

```diff
@@ -0,0 +1,100 @@
+require_relative "../test_helper"
+require_relative "../../lib/sneakers_handlers/exponential_backoff_handler"
+
+class SneakersHandlers::ExponentialBackoffHandlerTest < Minitest::Test
+  class FailingWorker
+    include Sneakers::Worker
+
+    from_queue "sneaker_handlers.exponential_back_test",
+      ack: true,
+      durable: false,
+      delays: [1, 3, 5],
+      exchange: "sneakers_handlers",
+      exchange_type: :topic,
+      routing_key: "lifecycle.created",
+      handler: SneakersHandlers::ExponentialBackoffHandler,
+      arguments: {
+        "x-dead-letter-exchange" => "sneakers_handlers.error"
+      }
+
+    def work(payload)
+      return reject!
+    end
+  end
+
+  def setup
+    cleanup!
+  end
+
+  def teardown
+    cleanup!
+  end
+
+  def test_handler_retries_with_ttl_retry_queues
+    exchange = channel.topic("sneakers_handlers", durable: false)
+    worker = FailingWorker.new
+
+    worker.run
+    exchange.publish("{}", routing_key: "lifecycle.created")
+
+    sleep 0.5
+
+    assert_equal 1, retry_queue(1).message_count
+
+    sleep 0.5
+    worker.run
+    sleep 0.5
+
+    assert_equal 1, retry_queue(3).message_count
+
+    sleep 2.5
+    worker.run
+    sleep 2.5
+
+    assert_equal 1, retry_queue(5).message_count
+
+    sleep 5.0
+    worker.run
+    sleep 0.1
+
+    assert_equal 1, error_queue.message_count
+  end
```

Comment:
Oh, cool. Good catch. I didn't realize that kept it running. Looks way better hahah